src/core/server/elasticsearch/elasticsearch_config.test.ts

```diff
@@ -311,11 +311,25 @@ describe('throws when config is invalid', () => {
 });
 
 describe('deprecations', () => {
+  it('logs a warning and rename `requestHeadersWhitelist` to `allowedRequestHeaders`', () => {
+    const { messages, migrated } = applyElasticsearchDeprecations({
+      requestHeadersWhitelist: ['foo', 'bar'],
+    });
+    expect(messages).toMatchInlineSnapshot(`
+      Array [
+        "\\"elasticsearch.requestHeadersWhitelist\\" is deprecated and has been replaced by \\"elasticsearch.allowedRequestHeaders\\"",
+      ]
+    `);
+    expect(migrated.elasticsearch).toEqual({
+      allowedRequestHeaders: ['foo', 'bar'],
+    });
+  });
+
   it('logs a warning if elasticsearch.username is set to "elastic"', () => {
     const { messages } = applyElasticsearchDeprecations({ username: 'elastic' });
     expect(messages).toMatchInlineSnapshot(`
       Array [
-        "Setting [${CONFIG_PATH}.username] to \\"elastic\\" is deprecated. You should use the \\"kibana_system\\" user instead.",
+        "Setting [elasticsearch.username] to \\"elastic\\" is deprecated. You should use the \\"kibana_system\\" user instead.",
```

Comment:
Updating the snapshots removed the var, however it felt unnecessary here, so I did not revert the change.